bokeh/models/math_text.py

```diff
@@ -0,0 +1,72 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2021, Anaconda, Inc., and Bokeh Contributors.
+# All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+''' Display a mathematics notation from a string value.
+'''
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+from __future__ import annotations
+
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# Bokeh imports
+from ..core.properties import NonNullable, String
+from ..model import Model
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+__all__ = (
+    'MathText',
+)
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+class MathText(Model):
+    """
+    Render mathematical content using `LaTeX <https://www.latex-project.org/>`_
+    notation.
+
+    .. note::
+        Bokeh uses `MathJax <https://www.mathjax.org>`_ to render text
+        containing mathematical notation.
+
+        MathJax only supports math-mode macros (no text-mode macros). You
+        can see more about differences between standard TeX/LaTeX and MathJax
+        here: https://docs.mathjax.org/en/latest/input/tex/differences.html
+    """
+
+    text = NonNullable(String, help="""
+    The text value to render as mathematical notation.
+    """)
+
+    def __init__(self, text: str = '', **kwargs) -> None:
+        if text:
+            kwargs['text'] = text
+
+        super().__init__(**kwargs)
```

Comment:
This entire initializer could be removed, I think? Initializers are only needed when something more than normal property handling is required. The ideal situation we aspire to for Bokeh models is to be purely declarative (i.e. no initializers)

Reply:
In this case we would use `MathText(text=r"\sin(x)")` instead of `MathText(r"\sin(x)")`?

Reply:
Yes, I think personally think that's fine, i.e. keep the model simple and standard, and layer on convenience in other ways, e.g with a `.accepts` clause  but would like to know what others think cc @bokeh/dev

Reply:
I opened a dedicated discussion to talk about convenience API for mathtext here https://github.com/bokeh/bokeh/discussions/11395